modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSConstants.java

```diff
@@ -320,7 +317,15 @@
     protected static final String PS_SELECT_EXPIRED_MESSAGES =
             "SELECT " + MESSAGE_ID + "," + DESTINATION_QUEUE
             + " FROM " + EXPIRATION_TABLE
-            + " WHERE " + EXPIRATION_TIME + "<" + System.currentTimeMillis();
+            + " WHERE " + EXPIRATION_TIME + "<" + System.currentTimeMillis()
+            + " AND " + MESSAGE_ID + ">=?"
+            + " AND " + DESTINATION_QUEUE + "=?";
+
+    protected static final String PS_SELECT_EXPIRED_MESSAGES_FROM_DLC =
+            "SELECT " + MESSAGE_ID + " FROM " + EXPIRATION_TABLE
+            + " WHERE " + EXPIRATION_TIME + "<" + System.currentTimeMillis()
```

Comment:
would this be correct ? it seems time will remain static after variable is inited (because you are putting System.currentTimeMillis()) into variable declaration.